collections/_article/dev-snapshot-godot-4-2-beta-1.md

```diff
@@ -0,0 +1,382 @@
+
+#### Shaders
+
+Visual shaders got some nice improvements with the addition of drop-down list properties to custom nodes ([GH-81688](https://github.com/godotengine/godot/pull/81688) by [Yuri Rubinsky](https://github.com/Chaosus)), and output ports for vector types are now expandable by default ([GH-82088](https://github.com/godotengine/godot/pull/82088) by [Dennis Manaa](https://github.com/DennisManaa), their first contribution).
+
+In text shaders, improvements were done to uniform sampler arrays to give them support for more hint types, such as filter modes ([GH-79100](https://github.com/godotengine/godot/pull/79100) by [Tomasz Chabora](https://github.com/KoBeWi)).
+
+
+### Scripting
+
+Godot has two main scripting languages, GDScript and C#, and each of them deserves special care and attention. That said, the new platform support coming to 4.2 is a very enticing reason to upgrade for C# users specifically!
+
+But let's start with a more general change. The script debugger now comes with full support for threaded code ([GH-76582](https://github.com/godotengine/godot/pull/76582) by [Juan Linietsky](https://github.com/reduz)), including the execution stack and breakpoints. This is something that has been requested since Godot 3 days, and now with the better multithreading support of Godot 4 it is finally possible to debug extra threads.
+
+<video autoplay loop muted playsinline title="A demo of new threading capabilities in the script debugger">
+  <source src="/storage/blog/godot-4-2-beta/threaded-debugger.mp4" type="video/mp4">
+</video>
+
+#### C# / .NET
+
+Back in March, Godot 4.0 was released replacing Mono support in favor of the new .NET framework, which created an unfortunate vacuum when it comes to mobile and web platform support. Godot C#/.NET maintainers had to wait for the new .NET 8.0 to continue their work and expand support to include those platforms. And so the time has come. Godot 4.2 features experimental support for Android (.NET 7.0+ required) and iOS (.NET 8.0 required) export targets.
+
+Thanks to [RedworkDE](https://github.com/RedworkDE) for [GH-73257](https://github.com/godotengine/godot/pull/73257) and thanks to [Andreia Gaita](https://github.com/shana) for [GH-82729](https://github.com/godotengine/godot/pull/82729)!
+
+There are still some kinks to iron out, some [caveats to consider](https://docs.godotengine.org/en/latest/tutorials/scripting/c_sharp/index.html#c-platform-support). And of course we rely on public testing to make sure everything works as expected. Until 4.3 this support will remain experimental, and we are still missing the web platform. But we are confident that the current state is good enough to plan ahead if your project is expected to ship to iOS or Android, so please start testing and report what you find!
```

Comment:
I feel like this sounds as if we're saying this will become stable in 4.3, but I don't think we can promise that.

The current iOS export relies on .NET 8 NativeAOT which is documented by Microsoft as experimental for this platform, so we probably should wait until they mark their stuff as stable to do the same thing ourselves.

However, we wanted to bring back some of the Mono embedding so if we can get that in time for 4.3 we may be able to remove the experimental label.

Reply:
It doesn't just sound like it, that's exactly what I wrote :) But if you think we can't commit to removing the experimental label in 4.3, I can rephrase it.

Reply:
I'd prefer to rephrase it so it doesn't sound so certain. I think in 4.3 we can remove the experimental label for Android, but iOS I don't know.